data-prepper-logstash-configuration/src/test/java/org/opensearch/dataprepper/logstash/mapping/OpenSearchPluginAttributesMapperTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.logstash.mapping;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+import org.opensearch.dataprepper.logstash.model.LogstashAttribute;
+import org.opensearch.dataprepper.logstash.model.LogstashAttributeValue;
+import org.opensearch.dataprepper.logstash.model.LogstashValueType;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasKey;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.opensearch.dataprepper.logstash.mapping.OpenSearchPluginAttributesMapper.LOGSTASH_OPENSEARCH_INDEX_ATTRIBUTE_NAME;
+
+class OpenSearchPluginAttributesMapperTest {
+
+    private static final String DATA_PREPPER_OPENSEARCH_INDEX_ATTRIBUTE = "index";
+
+    private OpenSearchPluginAttributesMapper createObjectUnderTest() {
+        return new OpenSearchPluginAttributesMapper();
+    }
+
+    @Test
+    void convert_missing_indexAttribute_to_return_empty_pluginSettings() {
+
+        final LogstashAttributesMappings logstashAttributesMappings = mock(LogstashAttributesMappings.class);
+        when(logstashAttributesMappings.getMappedAttributeNames()).thenReturn(Collections.emptyMap());
+
+        final Map<String, Object> pluginSettings = createObjectUnderTest()
+                .mapAttributes(Collections.emptyList(), logstashAttributesMappings);
+
+        assertThat(pluginSettings.size(), equalTo(0));
+        assertThat(pluginSettings, not(hasKey(DATA_PREPPER_OPENSEARCH_INDEX_ATTRIBUTE)));
+    }
+
+    @Test
+    void convert_emptyString_indexAttribute_to_return_pluginSettings_with_no_index_key() {
+
+        final LogstashAttribute logstashAttribute = createLogstashIndexAttribute("");
+
+        final LogstashAttributesMappings logstashAttributesMappings = mock(LogstashAttributesMappings.class);
+        when(logstashAttributesMappings.getMappedAttributeNames()).thenReturn(Collections.emptyMap());
+
+        final Map<String, Object> pluginSettings = createObjectUnderTest()
+                .mapAttributes(Collections.singletonList(logstashAttribute), logstashAttributesMappings);
+
+        assertThat(pluginSettings.size(), equalTo(1));
```

Comment:
I don't understand these assertions. There is one plugin setting here. But, it is not the `index`. What is it?

Reply:
`pluginSettings` is creating this: `pluginSettings: {null=}` where AttributeName is still `index` since that is a constant but AttributeValue is `""` so it's empty.